### ISSUE TITLE
Inconsistency in HDF5-formatted file reading depending on using .h5 or .hdf5 file extensions

### ISSUE THREAD

**Issue by zbarry** [flatironinstitute/CaImAn#326]
**Environment:**

Python 3.6, RHEL 6, terminal execution
Fresh clone of CaImAn master branch (commit 6bbade6)

**Issue:**

- Inconsistent behavior when importing .h5 vs. .hdf5 files when calling `motion_correction.py`
- When importing .hdf5, expected dataset in file is `mov`
- When importing .h5, expected dataset in file is `imaging`
- Other CaImAn functions expect the dataset to be `mov`

In addition, the check for presence of the dataset in `motion_correction.py` is only performed if the file is a .h5, not a .hdf5:

```
    elif extension == '.hdf5':
        with h5py.File(fname) as fl:
            T, d1, d2 = fl['mov'].shape

    elif extension == '.h5':
        with h5py.File(fname) as fl:
            if 'imaging' in fl.keys():
                T, _, d1, d2, _ = fl['imaging'].shape
            else:
                raise Exception(
                    'Unsupported file key for for h5 files in parallel motion correction')
```

Not sure if there's a reason for this. The motion correction through memmap saving works if the input is of .hdf5 extension.

**Error log:**

Provided the file ends in .h5 extension:

> Rigid Motion Correction
191.576
-56.3468600082
/data/mibs-inscopix/external/CaImAn-unmod/caiman/base/movies.py:270: UserWarning: ** Pixels averages are too negative. Removing 1 percentile. **
  '** Pixels averages are too negative. Removing 1 percentile. **')
/data/mibs-inscopix/external/CaImAn-unmod/caiman/base/movies.py:289: UserWarning: Pixels averages are too negative for template. Removing 1 percentile.
  'Pixels averages are too negative for template. Removing 1 percentile.')
cubic interpolation
-56.3468600082
cubic interpolation
-56.3468600082
cubic interpolation
Adding to movie 191.576
0
saving!
Traceback (most recent call last):
  File "/scratch/anaconda3/envs/inscopix/bin/caiman", line 11, in <module>
    load_entry_point('Inscopix-Analysis', 'console_scripts', 'caiman')()
  File "/data/mibs-inscopix/src/inscopix/run_caiman.py", line 395, in main
    args.func(args)
  File "/data/mibs-inscopix/src/inscopix/run_caiman.py", line 55, in stage_motion_correct
    max_tiff_frames=args.numtiffframes
  File "/data/mibs-inscopix/src/inscopix/caiman/motion_correct.py", line 39, in motion_correct
    save_movie=True
  File "/data/mibs-inscopix/external/CaImAn-unmod/caiman/motion_correction.py", line 512, in motion_correct_oneP_rigid
    mc.motion_correct_rigid(save_movie=save_movie, template=new_templ)
  File "/data/mibs-inscopix/external/CaImAn-unmod/caiman/motion_correction.py", line 243, in motion_correct_rigid
    border_nan=self.border_nan)
  File "/data/mibs-inscopix/external/CaImAn-unmod/caiman/motion_correction.py", line 2252, in motion_correct_batch_rigid
    use_cuda=use_cuda, border_nan=border_nan)
  File "/data/mibs-inscopix/external/CaImAn-unmod/caiman/motion_correction.py", line 2512, in motion_correction_piecewise
    'Unsupported file key for for h5 files in parallel motion correction')
Exception: Unsupported file key for for h5 files in parallel motion correction

Finally, note this discussion about determining if a dataset is present in an HDF5: h5py/h5py#726

I'm wondering if that might also cause these dataset checks to fail anyway, though not sure.

**Comment by epnev** [comment]
@zbarry 
I created PR #328 to deal with this. It should go through soon but feel free to check it out. Thanks for catching this.

**Comment by epnev** [comment]
merged into master, thanks again @zbarry
